src/test/FuzzerImpl.cpp

```diff
@@ -688,17 +763,87 @@ TransactionFuzzer::initialize()
         genOffersForPair(ASSET_B, ASSET_C, {13, 2, 14, 3, 15}, ltx);
         genOffersForPair(ASSET_C, ASSET_D, {6, 3, 7, 4, 8}, ltx);
 
-        // construct transaction
-        auto tx = createFuzzTransactionFrame(mSourceAccountID, ops,
-                                             mApp->getNetworkID());
+        attemptToApplyOps(ltx, mSourceAccountID, ops, *mApp);
+
+        ltx.commit();
+    }
+
+    {
+        LedgerTxn ltx(ltxOuter);
 
-        tx->attemptApplication(*mApp, ltx);
+        xdr::xvector<Operation> ops;
 
-        if (tx->getResultCode() != txSUCCESS)
+        // Reduce account balances so that fuzzing has a better change
+        // of exercising limits.
+        for (uint8_t i = 0; i < FuzzUtils::NUMBER_OF_PREGENERATED_ACCOUNTS; ++i)
         {
-            throw std::runtime_error("Error while initializing ledger state");
+            PublicKey account;
+            FuzzUtils::setShortKey(account, i);
+
+            // Reduce "account"'s native balance by paying the root.
+            LedgerKey accountKey;
+            accountKey.type(ACCOUNT);
+            accountKey.account().accountID = account;
+            LedgerEntry accountEntry = ltx.load(accountKey).current();
+            auto minBalance = getMinBalance(
+                mApp->getLedgerManager().getLastClosedLedgerHeader().header,
+                accountEntry.data.account());
+            auto remainingBalance =
+                minBalance +
+                FuzzUtils::DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY +
+                FuzzUtils::FUZZING_FEE *
+                    FuzzUtils::DEFAULT_NUM_TRANSACTIONS_TO_RESERVE_FEES_FOR;
+            assert(FuzzUtils::INITIAL_ASSET_DISTRIBUTION >= remainingBalance);
+            auto reduceNativeBalanceOp = txtest::payment(
+                mSourceAccountID, txtest::makeNativeAsset(),
+                FuzzUtils::INITIAL_ASSET_DISTRIBUTION - remainingBalance);
+            reduceNativeBalanceOp.sourceAccount.activate() =
+                toMuxedAccount(account);
+            ops.emplace_back(reduceNativeBalanceOp);
+
+            for (int j = 1; j <= FuzzUtils::NUMBER_OF_ASSETS_TO_ISSUE; ++j)
+            {
+                auto const asset = FuzzUtils::makeAsset(j);
+                if ((i != 0) && (i != j))
+                {
+                    PublicKey issuer;
+                    FuzzUtils::setShortKey(issuer, j);
+
+                    // Reduce "account"'s balance of asset "j" by paying the
+                    // issuer.
+                    LedgerKey trustLineKey;
+                    trustLineKey.type(TRUSTLINE);
+                    trustLineKey.trustLine().accountID = account;
+                    trustLineKey.trustLine().asset = asset;
+                    LedgerEntry trustLineEntry =
+                        ltx.load(trustLineKey).current();
+                    auto extVersion = trustLineEntry.data.trustLine().ext.v();
+                    auto sellingLiabilities =
```

Comment:
You're working too hard. Use `getAvailableBalance`.

Reply:
Done.